### PR TITLE
performance improvements to history

### PR DESCRIPTION
**Pull request by richorama**
```
                                Method | SiloCount | GrainTypeCount | GrainMethodCount | HistorySize |     Mean |     Error |    StdDev |
-------------------------------------- |---------- |--------------- |----------------- |------------ |---------:|----------:|----------:|
            Test_QueryAll_TraceHistory |        10 |             50 |               10 |         100 | 52.65 ms | 1.3421 ms | 1.8371 ms |
           Test_QuerySilo_TraceHistory |        10 |             50 |               10 |         100 | 18.20 ms | 0.1283 ms | 0.1200 ms |
          Test_QueryGrain_TraceHistory |        10 |             50 |               10 |         100 | 12.22 ms | 0.0981 ms | 0.0766 ms |
 Test_GroupByGrainAndSilo_TraceHistory |        10 |             50 |               10 |         100 | 84.89 ms | 1.3049 ms | 1.0188 ms |
```

### CODE REVIEW

**Comment by SebastianStehle** [comment]
Can you also add a benchmark for the Add method?

File: OrleansDashboard/History/TraceHistory.cs [hunk 1]
```diff
@@ -96,14 +97,15 @@ public void Add(DateTime now, string siloAddress, SiloGrainTraceEntry[] grainTra
 
             foreach (var historicValue in history.Where(x => x.Grain == grain))
             {
-                var grainMethodKey = $"{grain}.{historicValue.Method}";
```
Comment (SebastianStehle) [inline OrleansDashboard/History/TraceHistory.cs]: Should it not be faster to precalculate this value in the Add() method?

**Comment by richorama** [comment]
yes, I am planning on benchmarking the add method, it's a little more complicated, but I hope to add that today.

File: OrleansDashboard/History/TraceHistory.cs (hunk 1 above)
Comment (richorama) [reply OrleansDashboard/History/TraceHistory.cs]: possibly, worth benchmarking :¬)

I found that the DateTime.ToString was _very_ slow.

**Pull request merged**